pkg/transaction_details/transaction_details.go

```diff
@@ -1,18 +1,54 @@
 package transaction_details
 
 import (
+	"fmt"
 	"github.com/eucrypt/unmarshal-go-sdk/pkg/constants"
 	"github.com/eucrypt/unmarshal-go-sdk/pkg/transaction_details/types"
 )
 
 type TokenTxnsOpts struct {
 	Contract string
+	PaginationOptions
+}
+type PaginationOptions struct {
 	Page     int
 	PageSize int
 }
 
+type TransactionDetailsOpts struct {
+	PaginationOptions
+	FromBlock uint64
+	ToBlock   uint64
+}
+
+const RawFormat = "raw"
+
+func (p PaginationOptions) getPaginationAsQueryParams() map[string]interface{} {
+	var queryParams = map[string]interface{}{
+		"page":     fmt.Sprint(p.Page),
+		"pageSize": fmt.Sprint(p.PageSize),
+	}
+	return queryParams
+}
+
+func (options TokenTxnsOpts) getMappableQueryParams() map[string]interface{} {
+	var queryParams = options.getPaginationAsQueryParams()
+	queryParams["contract"] = options.Contract
```

Comment:
Why not have `from` and `to` block params here?

Reply:
Also, `format` could just become an argument to this method right?

Reply:
I added the missing block limits support